api/filtering/db_custom_filters.py

```diff
@@ -158,6 +169,7 @@ def build_operating_system_filter(filter_param: dict) -> tuple:
             os_filter_list.append(os_field.astext.operate(comparator, None))
 
         elif comparison.comparator == "eq":
+            print("~~~~~~~~~~~~~~~~~~~``", os_field["name"].astext == comparison.name)
```

Comment:
Around here is where you'd convert the OsComparison to an actual DB operation. If we went with the logic I posted above, this next section could be something like this:

```suggestion
            os_filters = [
                os_field["name"].astext == comparison.name,
            ]
            if comparison.major is not None:
                os_filters.append(os_field["major"].astext.cast(Integer) == comparison.major)
                if comparison.minor:
                    os_filters.append(os_field["minor"].astext.cast(Integer) == comparison.minor)
            
            os_filter_list.append(and_(*os_filters))
```
Of course, you'll need to handle "neq" as well :)